src/wp-includes/class-wp-scripts.php

```diff
@@ -648,7 +681,18 @@ private function should_delay_inline_script( $handle, $position ) {
 		 * dependency's after inline script.
 		 */
 		foreach ( $deps as $dep ) {
-			if ( $this->is_delayed_strategy( $this->get_eligible_loading_strategy( $dep ) ) ) {
+			if ( ! array_key_exists( $dep, $this->registered ) ) {
+				continue;
+			}
+
+			// If the dependency is an alias, look at its members.
+			if ( ! $this->registered[ $dep ]->src ) {
+				foreach ( $this->registered[ $dep ]->deps as $alias_dep ) {
```

Comment:
Just in case there is nested aliasing going on, perhaps this would be more robust:

```suggestion
				foreach ( $this->get_unaliased_deps( $this->registered[ $dep ]->deps ) as $alias_dep ) {
```